### DISCUSSION TITLE
Get user preference when using `ui.dark_mode(None)`

### DISCUSSION THREAD

**Discussion by marlonmueller** [zauberzeug/nicegui#2415] [Ideas / Feature Requests]
Hi,

In my application, I use `ui.dark_mode(None)` to dynamically set the dark mode based on the client's system preference. 

I noticed that the `ui.aggrid` theme does not automatically change along with the dark mode setting. In an attempt to set the theme manually, I expected `dark.value` to be either `True` or `False` after setting `dark = ui.dark_mode(None)`. However, it currently is just `None`.

Thanks.

**Comment by falkoschindler** [comment]
Hi @MarlonMueller,

`ui.dark_mode` represents the "server's choice", i.e. what the application thinks the mode should be. If it should be up to the user, the value is set to `None`.

Unfortunately it's not so easy to know which mode the user is using. Imagine the dark mode being activated in the operation system after the user opened the NiceGUI page. The browser would need to notice and inform the server about the changing user preference. Currently this is not happening.

[There is a related discussion](https://github.com/zauberzeug/nicegui/discussions/2178) about how to get the current user preference via a JavaScript call. But I'm aware that this isn't ideal for updating `ui.aggrid` automatically.

**Comment by marlonmueller** [comment]
Whilst it might be a nice feature, it thus seems like implementing this (browser informs server) could introduce other unexpected behaviour, so the JS solution it is! Thanks for your help (also @rodja as he asked me to create the feature request)! 👍🏼
